src/admin.py

```diff
@@ -339,20 +340,111 @@ def list_workflow_templates():
+#         """Parses the retention time string (e.g., '10D', '2W', '1M') into a timedelta."""
+#         match = re.match(r"(\d+)([mhDWMY])", retention)
+#         if not match:
+#             raise typer.BadParameter("Invalid retention time format. Use <number>[D|W|M].")
+#         value = int(match.group(1))
+#         unit = match.group(2)
+#         if unit == "m":
+#             return timedelta(minutes=value)
+#         if unit == "h":
+#             return timedelta(hours=value)
+#         elif unit == "D":
+#             return timedelta(days=value)
+#         elif unit == "W":
+#             return timedelta(weeks=value)
+#         elif unit == "M":
+#             # Approximate months to 30 days for simplicity in CLI
+#             return timedelta(days=value * 30)
+#         elif unit == "Y":
+#             # Approximate years to 365 days for simplicity in CLI
+#             return timedelta(days=value * 365)
+#         raise typer.BadParameter("Invalid retention time unit.")
+
+#     db = database.SessionLocal()
+#     deleted_files_query = db.query(File)
+#     deleted_files_query._include_deleted = True
+#     deleted_files_query = deleted_files_query.filter(
+#         File.is_deleted == True, File.is_purged == False
+#     )
+
+#     if retention:
+#         try:
+#             retention_delta = _parse_retention_time(retention)
+#             cutoff_time = datetime.now(timezone.utc) - retention_delta
+#             deleted_files_query = deleted_files_query.filter(File.deleted_at <= cutoff_time)
+#         except typer.BadParameter as e:
+#             print(f"[bold red]Error:[/bold red] {e}")
+#             return
+
+#     deleted_files = deleted_files_query.all()
+
+#     if not deleted_files:
+#         print("[bold green]No files are marked as deleted.[/bold green]")
+#         return
+
+#     total_size = sum(file.filesize for file in deleted_files if file.filesize)
+#     num_files = len(deleted_files)
+
+#     print("[bold blue]Delete request summary:[/bold blue]")
+#     print(f"  Number of files to delete: [bold]{num_files}[/bold]")
+
+#     def format_size(size_bytes):
+#         """Formats bytes into a human-readable string."""
+#         if size_bytes is None:
+#             return "N/A"
+#         units = ["bytes", "KB", "MB", "GB", "TB", "PB"]
+#         i = 0
+#         while size_bytes >= 1024 and i < len(units) - 1:
+#             size_bytes /= 1024
+#             i += 1
+#         return f"{size_bytes:.2f} {units[i]}"
+
+#     print(f"  Total size: [bold]{format_size(total_size)}[/bold]")
+
+#     if not force and not typer.confirm(f"Are you sure you want to delete {num_files} files?"):
+#         print("[bold red]Deletion cancelled.[/bold red]")
+#         return
+
+#     print("[bold yellow]Deleting files...[/bold yellow]")
+#     for file in deleted_files:
+#         file.purge(db)
+#         print("  Deleted file:", file.path)
+
+
 @app.command()
 def purge_deleted_files(
```

Comment:
This method is massive. We should probably see if it's possible to break it down into smaller methods somehow to make it easier to understand / modify (this can be in another PR).

Reply:
I agree, it grew a bit more than anticipated. I want to refactor this admin script in general. I make a note of it. And also add tests....